media/server/gstplayer/source/GstGenericPlayer.cpp

```diff
@@ -1467,7 +1467,8 @@ bool GstGenericPlayer::getVolume(double &currentVolume)
     }
 
     GstElement *sink{getSink(MediaSourceType::AUDIO)};
-    if (sink && m_glibWrapper->gObjectClassFindProperty(G_OBJECT_GET_CLASS(sink), "fade-volume"))
+    if (m_context.audioFadeEnabled && sink &&
+        m_glibWrapper->gObjectClassFindProperty(G_OBJECT_GET_CLASS(sink), "fade-volume"))
```

Comment:
If speed is an issue then wouldn't it be quicker to check this once...
 m_glibWrapper->gObjectClassFindProperty(G_OBJECT_GET_CLASS(sink), "fade-volume")
 ...before setting m_context.audioFadeEnabled (this is set in two places, and the property is already checked for in one of these two places. We'd just need to add a check before the other location too, and perhaps add a comment to the header file to say that the existence of the property must be verified before setting the variable)

Reply:
I don't agree, in setVolume, where audioFadeEnabled is set, "audio-fade property" is checked. Here we check "fade-volume", so it's different property. gObjectClassFindProperty is fast, on broadcom it always returns true. Slow operation, that we want to avoid when audio-fade is disabled is: gObjectSet(audioSink, "audio-fade", fadeStr, nullptr);

Reply:
Unfortunately it's not possible to check if fade-volume is present in else if in set volume - sink may be null there, so code has to stay as it is now